website/uc-doc/system/wazo-auth/developer.md

```diff
@@ -291,3 +291,282 @@ email_notification_plugin: proxy
 proxy_confirmation_url: confirmation.example.com
 proxy_password_reset_url: password_reset.example.com
 ```
+
+### IdP plugins
+
+> **Note**
+> IdP comes from an abbreviation of "Identity Provider"
+
+The `wazo_auth.idp` entrypoint namespace is supported to register alternative authentication
+mechanisms to those supported by default.
+
+An IdP plugin can implement a specific mechanism of authentication for some login requests. This
+would allow, for example, checking credentials from a login request by querying a third party API.
+
+This plugin interface is complemental to the [`wazo_auth.backend` plugin interface](Backends), as
+the authentication mechanism of an IdP must result in the selection of an appropriate
+`wazo_auth.backend` implementation, along with the wazo user identity(wazo username or email
+address) which is being authenticated.
+
+- IdP plugin interface deals with authentication at a more abstract level, by leaving the plugin to
+  process the login request, without imposing expectations such as the presence of a
+  username/password credential; an IdP plugin authentication mechanism may rely on a
+  `wazo_auth.backend` `verify_password` implementation, or not;
+- `wazo_auth.backend` plugins are used both to authenticate the username/password credential of a
+  login request, as well as to provide the ACLs & metadata of the generated access token; IdP
+  plugins cannot affect the ACLs & metadata resulting from a successful login other than by
+  selecting the appropriate `wazo_auth.backend` implementation;
+- an IdP plugin can elect to handle a login request, and can choose(statically or dynamically) which
+  `wazo_auth.backend` implementation to use in the authentication process; this enables new
+  `wazo_auth.backend` implementations to be used without modifying the core wazo-auth code.
+
+#### Interface
+
+The plugin interface is defined in the wazo-auth codebase as a python class.
+
+https://github.com/wazo-platform/wazo-auth/blob/master/wazo_auth/interfaces.py#L114-L147
```

Comment:
How about using a link to a specific version of wazo-auth?
https://github.com/wazo-platform/wazo-auth/blob/186c3d8aad948e829904ef8b70c05cf4b636e58f/wazo_auth/interfaces.py#L114-L147

Reply:
Yeah, makes sense. I was thinking "if the code changes on master, someone can notice and point out the discrepancy, but if I permalink a specific commit the doc might become out-of-date and nobody will notice from this doc".

Reply:
Fixed